BeerProject/src/main/java/wawer/kamil/beerproject/service/impl/BeerServiceImpl.java

```diff
@@ -0,0 +1,191 @@
+package wawer.kamil.beerproject.service.impl;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+import wawer.kamil.beerproject.model.Beer;
+import wawer.kamil.beerproject.model.Brewery;
+import wawer.kamil.beerproject.exceptions.InvalidImageParameters;
+import wawer.kamil.beerproject.exceptions.NoContentException;
+import wawer.kamil.beerproject.repositories.BeerRepository;
+import wawer.kamil.beerproject.repositories.BreweryRepository;
+import wawer.kamil.beerproject.service.BeerService;
+import wawer.kamil.beerproject.utils.upload.ImageUpload;
+
+import java.io.IOException;
+import java.util.List;
+
+
+@Service
+@AllArgsConstructor
+@Slf4j(topic = "application.logger")
+public class BeerServiceImpl implements BeerService {
+
+
+    private final BeerRepository beerRepository;
+    private final BreweryRepository breweryRepository;
+    private final ImageUpload imageUpload;
+
+    private static final String THE_BREWERY_BASE_ON_ID_HAS_NOT_BEEN_FOUND = "The brewery base on id: {} has not been found";
+    private static final String THE_BEER_BASE_ON_ID_HAS_NOT_BEEN_FOUND = "The beer base on id: {} has not been found";
+
+    //get beers
+
+    @Override
+    public Page<Beer> findAllBeersPage(Pageable pageable) {
+        return beerRepository.findAll(pageable);
+    }
+
+    @Override
+    public List<Beer> findAllBeersList() {
+        return beerRepository.findAll();
+    }
+
+    @Override
```

Comment:
Jeżeli twoje wyjątki będą rozszerzeniem klasy `RuntimeException` a nie `Exception` to nie będziesz musiał ich rzucać wyżej a i tak będą obsługiwane w Advicerze - kod będzie zdecydowanie czytelniejszy